ui/app/AppLayouts/Chat/views/ChatColumnView.qml

```diff
@@ -186,19 +178,17 @@ Item {
                     delegate: Repeater {
                         model: subItems
                         delegate: ChatContentView {
+                            store: rootStore
                             Component.onCompleted: {
-                                parentModule.prepareChatContentModuleForChatId(model.itemId)
-                                chatContentModule = parentModule.getChatContentModule()
+                                rootStore.chatCommunitySectionModule.prepareChatContentModuleForChatId(model.itemId);
+                                rootStore.chatContentModule = rootStore.currentChatContentModule();
                             }
                         }
                     }
                 }
                 DelegateChoice { // In all other cases
                     delegate: ChatContentView {
-                        Component.onCompleted: {
```

Comment:
@alexandraB99 another issue I see here is adding `chatContentModule` to the store which is unique per chat/community section. That's why I deliberately removed `rootStore`, and left  `chatContentModule` as a property of the `ChatContentView` component. 

I agree that it should be part of a store, but a part of a store maintained per `ChatContentView` (which we don't have yet), not part of a store maintained per `ChatLayout`. This is a big thing.

And that's what I was pointing to that I said that some properties of the children stores will be defined from the parent store, cause they are known only in the runtime of the app, not before that.

So in our case it should be something like this (pseudo code just for the sake of easier reading):
```
// ChatContentView.qml
ChatContentView {
    id: chatContentId
    chatContentStore: ChatContentRootStore {
        property var chatContentModule
    }
}


// ChatLayout.qml
ChatLayout {
    id: chatLayoutId
    sectionStore: SectionRootStore {
    }

    // Now if we have here Message component it will be set from the store of this component
    Repeater {
        model: someModel
        delegate: ChatContentView {
             chatContentStore.chatContentModule: {
                 chatLayoutId.sectionStore.chatCommunitySectionModule.prepareChatContentModuleForChatId(model.itemId)
                 return chatLayoutId.sectionStore.chatCommunitySectionModule.getChatContentModule()
             }
        }
    }
  }
```


The same states for the `MessageStore` it's a part of the app's root store, but in no case it can be there. Message store should be part of the `SectionRootStore`, since each chat/channel has its own `SectionRootStore`.

Reply:
> @alexandraB99 why is everything here removed? That means that we don't have `chatContentModule` appropriately set for all chats in case of Chat section and for all channels which are not belonging to any of the category in case of Community section, what we don't want actually. If you take a closer look here, this is a Repeater dealing with a model where each item may have or may not have subItems. Based on that DelegateChooser is used on the role `type`, that further means if an item doesn't have subItems, create `ChatContentView`, otherwise if an item has subitems, create `ChatContentView` for each of item of subItems.
> 
> So please add here this:
> 
> ```
> Component.onCompleted: {
>     rootStore.chatCommunitySectionModule.prepareChatContentModuleForChatId(model.itemId)
>     chatContentModule = rootStore.chatCommunitySectionModule.getChatContentModule()
> }
> ```
> 
> **Notice that I deliberately removed `rootStore` from `rootStore.chatContentModule`, and left it as it was, will explain why in the comment bellow.**

Please see comment above!

Reply:
> @alexandraB99 another issue I see here is adding `chatContentModule` to the store which is unique per chat/community section. That's why I deliberately removed `rootStore`, and left `chatContentModule` as a property of the `ChatContentView` component.
> 
> I agree that it should be part of a store, but a part of a store maintained per `ChatContentView` (which we don't have yet), not part of a store maintained per `ChatLayout`. This is a big thing.
> 
> And that's what I was pointing to that I said that some properties of the children stores will be defined from the parent store, cause they are known only in the runtime of the app, not before that.
> 
> So in our case it should be something like this (pseudo code just for the sake of easier reading):
> 
> ```
> // ChatContentView.qml
> ChatContentView {
>     id: chatContentId
>     chatContentStore: ChatContentRootStore {
>         property var chatContentModule
>     }
> }
> 
> 
> // ChatLayout.qml
> ChatLayout {
>     id: chatLayoutId
>     sectionStore: SectionRootStore {
>     }
> 
>     // Now if we have here Message component it will be set from the store of this component
>     Repeater {
>         model: someModel
>         delegate: ChatContentView {
>              chatContentStore.chatContentModule: {
>                  chatLayoutId.sectionStore.chatCommunitySectionModule.prepareChatContentModuleForChatId(model.itemId)
>                  return chatLayoutId.sectionStore.chatCommunitySectionModule.getChatContentModule()
>              }
>         }
>     }
>   }
> ```
> 
> The same states for the `MessageStore` it's a part of the app's root store, but in no case it can be there. Message store should be part of the `SectionRootStore`, since each chat/channel has its own `SectionRootStore`.

Once again, there is no point to create an instace for each and ALL chats, communities, categories. Currently the `prepareChatContentModuleForChatId` is called every time a chat is selected and based on it's id, retrieving this way the chat content on demand. Do you see any reason for keeping so many instances alive during runtime?

Reply:
@alexandraB99 I already answered [here](https://github.com/status-im/status-desktop/pull/4316/files#r771809876).